RevenueCatUI/CustomerCenter/Data/LoadPromotionalOfferUseCase.swift

```diff
@@ -84,22 +84,58 @@ private extension LoadPromotionalOfferUseCase {
     }
 
     private func findDiscount(
-        for product: StoreProduct,
+        for activeProduct: StoreProduct,
         productIdentifier: String,
         promoOfferDetails: CustomerCenterConfigData.HelpPath.PromotionalOffer
```

Comment:
Is there a case where the `activeProduct` is not the one defining the `productIdentifier`?

Reply:
yes, the productIdentifier could be of the target product. So active product could be `monthly` and productIdentifier could be `annual`. I will change `productIdentifier` to `discountedProductIdentifier`

Reply:
no, it's always `activeProduct.productIdentifier`. I will remove it